iotile_ext_cloud/iotile/cloud/cloud.py

```diff
@@ -161,9 +161,8 @@ def check_time(self, max_slop=300):
 
     @param("device_id", "integer", desc="ID of the device that we want information about")
     @param("new_sg", "string", desc="The new sensor graph id that we want to load")
-    def set_sensorgraph(self, device_id, new_sg):
+    def set_sensorgraph(self, device_id, new_sg, app_tag=None):
```

Comment:
Where are you using this?

You need to do the same

```
try:
     sg = self.api.sg(new_sg).get()
except RestHttpBaseException, exc:
     ...
```

to get the sensor graph and then check that `sg['app_tag'] == app_tag`